src/components/Questionnaire/QuestionnaireForm.tsx

```diff
@@ -585,11 +585,17 @@ export function QuestionnaireForm({
               q.structured_type as keyof typeof STRUCTURED_TYPE_VALIDATORS
             ];
 
+          // Skip appointment validator if question is not required
           if (validator) {
-            const validationErrors = validator(response?.values?.[0], q.id);
-            errors.push(...validationErrors);
-            if (validationErrors.length > 0) {
-              firstErrorId = firstErrorId ? firstErrorId : q.id;
+            // Only validate appointment fields if the question is required
+            if (q.structured_type === "appointment" && !q.required) {
```

Comment:
Let's change the condition instead to:

```suggestion
            if (q.structured_type !== "appointment || (q.structured_type === "appointment" && q.required)) {
```

So we wouldn't need the else.

Reply:
Also complete the merge checklist